src/static_canvas.class.js

```diff
@@ -871,16 +841,34 @@
         fabric.util.clipContext(this, canvasToDrawOn);
       }
 
-      this._renderBackground(canvasToDrawOn);
-      this._renderObjects(canvasToDrawOn, activeGroup);
-      this._renderActiveGroup(canvasToDrawOn, activeGroup);
+      if (activeGroup && !this.preserveObjectStacking) {
+        for (var i = 0, length = this._objects.length; i < length; i++) {
+          object = this._objects[i];
+          if (!activeGroup.contains(object)) {
+            objsToRender.push(object);
+          }
+          else {
+            activeGroupObjects.push(object);
+          }
+        }
+        activeGroup._set('_objects', activeGroupObjects);
+      }
+      else {
+        objsToRender = this._objects;
+      }
 
+      //apply viewport transform once for all rendering process
+      canvasToDrawOn.setTransform.apply(canvasToDrawOn, this.viewportTransform);
+      this._renderBackground(canvasToDrawOn);
+      this._renderObjects(canvasToDrawOn, objsToRender);
+      !this.preserveObjectStacking && this._renderObjects(canvasToDrawOn, [activeGroup]);
```

Comment:
Also could be `this.preserveObjectStacking || this._renderObjects(canvasToDrawOn, [activeGroup]);`